src/ducks/zkpr.ts

```diff
@@ -229,23 +219,29 @@ export const useIdCommitment = () => {
   }, deepEqual);
 };
 
+interface ZKPRListener {
+  logout: () => void;
+  identityChanged: (idCommitment: string) => void; // TODO: CK data to become: { idCommitment: string; provider: string }
+}
+
 export class ZKPR {
   private client: any;
 
   constructor(client: any) {
     this.client = client;
   }
 
-  on(eventName: string, cb: (data: unknown) => void) {
-    return this.client.on(eventName, cb);
+  on<U extends keyof ZKPRListener>(event: U, listener: ZKPRListener[U]) {
```

Comment:
@isk maybe consider reusing something like this to improve the typing of `on` in the cryptkeeper repo too?

Reply:
ideally we should extract this entire class into an npm module, used by any devs who wants to integrate with crypt keeper

Reply:
Yes I agree it would be valuable to have a package for the crypt keeper client/api

Reply:
Yes we discussed yesterday that we need to support several packages: 
- `@types/cryptkeeper` for supporting the types
- `@cryptkeeper/provider`